modules/integration/tests-common/integration-test-utils/src/main/java/org/wso2/appserver/integration/common/utils/PasswordEncryptionUtil.java

```
@@ -0,0 +1,143 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.appserver.integration.common.utils;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.NodeList;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.xpath.XPath;
+import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathFactory;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+
+
+/**
+ * This Class is implemented to encrypt password and check password has encrypted.
+ */
+
+public class PasswordEncryptionUtil {
+
+    private static final Log log = LogFactory.getLog(PasswordEncryptionUtil.class);
+
+    /**
+     * By checking the master-datasources.xml password node has encrypted attribute
+     *
+     * @return boolean - true : if password has encrypted ,false : if not
+     * @throws Exception - Error when getting and reading master-datasources.xml
+     */
+    public static boolean checkIsPasswordEncrypted(String carbonHome) throws Exception {
```

Throwing General exception is not recommended. You need to throw the specific exception here